File: boj/8958.cpp
#include <iostream>

using namespace std;

// https://www.acmicpc.net/problem/8958
// gcc version 6.3.0 (MinGW.org GCC-6.3.0-1)

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    int tc;
    cin >> tc;

    while(tc--) {
        string s;
        cin >> s;

        int score[s.length()] = {0};
        int sum = 0;

        for(int i=0; i<s.length(); i++) {
            if(s[i] == 'O') {
                if(i == 0)
                    score[i] = 1;
                else
                    score[i] = score[i-1] + 1;
            }
            sum += score[i];
        }

        cout << sum << '\n';
    }

    return 0;
}

File: BOJ/8958.java
import java.util.*;
import java.io.*;

// https://www.acmicpc.net/problem/8958

class Main {
    static int stoi(String s) {
        return Integer.parseInt(s);
    }

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int n = stoi(br.readLine());

        for (int i = 0; i < n; i++) {
            String s = br.readLine();
            int len = s.length();

            int sum = 0;
            int score = 1;
            /**
             * O 면 더한 후에 score++
             * X 면 더하지 않고 score = 1
             */
            for (int j = 0; j < len; j++) {
                if (s.charAt(j) == 'O') {
                    sum += score;
                    score++;
                } else
                    score = 1;
            }

            sb.append(sum + "\n");
        }

        System.out.print(sb);
    }
}